pkg/completion/doc.go

```diff
@@ -55,4 +55,10 @@
 // Wait blocks until either all Completions are completed, or the context is
 // canceled (e.g. times out), whichever happens first. The returned error is
 // non-nil in the case the context is canceled, nil otherwise.
+//
+// A Completion can also be created with a callback, which is called at most
+// once when the Completion is successfully completed before the context is
+// cancelled:
+//
+//    comp := wg.AddCompletionWithCallback(func() { fmt.Println("completed') })
```

Comment:
Should mention in which goroutine/context the callback is called, and/if when it makes sense to have the callback start it's own goroutine?

Reply:
Good idea. I added this sentence:
```
// The callback is called in the goroutine which calls Complete the first time.
```